assets/js/base/components/block-error-boundary/block-error.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { WC_BLOCKS_ASSET_URL } from '@woocommerce/block-settings';
+import PropTypes from 'prop-types';
+
+const BlockError = ( {
+	imageUrl = `${ WC_BLOCKS_ASSET_URL }img/block-error.svg`,
+	header = __( 'Oops!', 'woo-gutenberg-products-block' ),
+	content = (
+		<p>
+			{ __(
+				'There was an error with loading this content.',
+				'woo-gutenberg-products-block'
+			) }
+		</p>
+	),
```

Comment:
Hmm... if you're going to do this, what about just making `BlockError` a balanced element.  So to pass in the content, implementation would be:

```js
<BlockError>
    <p>{ 'There was an error with loading this content.' } </p>
</BlockError>
```

Then you'd just detect in the `BlockError` if there are `children` and if not, add the default?  That eliminates the need for a n additional prop here.

The other consideration though is whether we want to explicitly only allow text here (which I think may be important).  The reason being is that any Exceptions/Errors in the Error boundary itself are _not_ caught (which would include this component because it's used directly in the ErrorBoundary.  So I _think_ it would be preferable here to just accept text.  This should be a fairly restrictive api (and still gives us room to iterate on this later if we find need for more flexibility).

Reply:
> if you're going to do this, what about just making BlockError a balanced element.

My main reason against that was that it seemed weird to me to consider the line of text a child while not considering the heading too. (Anyway, ignore this discussion because... :point_down: ).

> The other consideration though is whether we want to explicitly only allow text here (which I think may be important). The reason being is that any Exceptions/Errors in the Error boundary itself are not caught (which would include this component because it's used directly in the ErrorBoundary. So I think it would be preferable here to just accept text. This should be a fairly restrictive api (and still gives us room to iterate on this later if we find need for more flexibility).

Good point! I had in mind that in some cases it might be interesting to add more than one line of text or adding a list of "things to do to fix this issue", but I guess we can iterate on this component in the future if we see this is a necessity, so I made it so it only accepts text as it was before (6d3d1bb7c803323129d4fa7b2d040c7e4c3e5aea).